packages/fei/base/fei_EqnCommMgr.cpp

```diff
@@ -741,9 +741,9 @@ void EqnCommMgr::exchangeSoln()
 }
 
 //------------------------------------------------------------------------------
-// This works around an issue with the ARMHPC 20.1 compiler
+// This works around an issue with the clang and ARMHPC compiler
 // Needs to be revisited with later versions
-#ifdef __ARM_HPC_COMPILER_VERSION__
+#if defined (__clang__) && ! defined(__INTEL_CLANG_COMPILER)
```

Comment:
would we need to have if defined ARM or if defined clang?  Or is straight replacement ok?

Reply:
@mwglass

Reply:
The ARMHPC compiler defines __clang__ so we don't need the __ARM_HPC_COMPILER_VERSION__ check any longer. The previous check was to just restrict this workaround for the ARMHPC compiler but now this looks to be an issue with the generic clang compiler too but does not affect the Intel next gen compiler.